mqm_tool.py

```diff
@@ -0,0 +1,391 @@
+                        elif in_grids[grid_array_ind][3] != initial_bb[3] and in_grids[grid_array_ind][2] != initial_bb[2]:
+                            if (np_array[coordinate_ind, :][0] >= in_grids[grid_array_ind][0] and
+                                np_array[coordinate_ind, :][0] < in_grids[grid_array_ind][2] and
+                                np_array[coordinate_ind, :][1] >= in_grids[grid_array_ind][1] and
+                                np_array[coordinate_ind, :][1] < in_grids[grid_array_ind][3]):
+                                tmp_array[grid_array_ind] += 1                
+        
+                for grid_array_ind in range(len(in_grids)):
+                    if tmp_array[grid_array_ind] > 0:
+                        road_counts_his[grid_array_ind] += 1
+        # ==========================================
+        for array_index in range(len(in_grids)):
+            csv_matrix.append([array_index + 1, in_counts[array_index], road_counts_his[array_index]])
+
+        # write out a csv file
+        util = Utility()
+        util.csv_writer(csv_matrix, os.path.join(out_folder, 'road-' + os.path.basename(out_folder) + '.csv'))
+        
+
+def get_argument():
+    """ Argument grabbing function.
+
+    This function grabs all of the arguments that the program needs.
+    
+    Returns:
+        args.folderPath: an input folder path.
+        args.maxDepth: a maximum tree depth.
+        output_folder: a result folder.
+        int(args.countNum): a count number for a stop condition in the first k-d tree.
+        float(args.gridPercent): a percetage value for grids.
+        max_count: a maximum count in the second k-d tree.
+        path: a path to a histogram folder.
+        geojson_path: a path to a geojson folder.
+        
+    """
+    # declare arguments and variables
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--folderPath', type = str, default='', help='path to an input folder')
+    parser.add_argument('--maxDepth', type = str, default='10', help='max depth of a k-d tree')
+    parser.add_argument('--countNum', type = str, default='10', help='a count value for a stop condition')
+    parser.add_argument('--gridPercent', type = str, default='0.9', help='a grid percentage')
+    parser.add_argument('--maxCount', type = str, default='', help='maximum count to the second k-d tree')
+    args = parser.parse_args()
+    max_count = -1
+    path = 'histogram'
+    geojson_path = 'geojson'
+    output_folder = os.path.join(os.path.split(args.folderPath)[0], 'result')
+
+    
+    if args.maxCount:
+        max_count = int(args.maxCount)
+
+    
+    return args.folderPath, args.maxDepth, output_folder, int(args.countNum), float(args.gridPercent), max_count, path, geojson_path
+
+
+def stop_condition(count_zero_list, count_list, grid_percent, count_num, cell_num, out_distribution):
+    """ Stop condition function.
+
+    This function calculates the stop condition to the first k-d tree.
```

Comment:
It would be great to have a description of the condition here.

Reply:
added the descriptive comments to this function.